libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+    fsx.ensureDirSync(folderPath);
+  }
+
+  /**
+   * Reads the contents of the folder, not including "." or "..".
+   * Behind the scenes it uses `fs.readdirSync()`.
+   * @param folderPath - The absolute or relative path to the folder which should be read.
+   * @param options - Optional settings that can change the behavior. Type: `IReadFolderOptions`
+   */
+  public static readFolder(folderPath: string, options?: IReadFolderOptions): Array<string> {
+    options = {
+      absolutePath: false,
+      ...options
+    };
+
+    if (!FileSystem.exists(folderPath)) {
+      throw new Error(`Cannot read contents of folder, as it does not exist: "${folderPath}"`);
+    }
+
+    const fileNames: Array<string> = fsx.readdirSync(folderPath);
+
+    if (options.absolutePath) {
+      return fileNames.map(fileName => pathUtilities.resolve(folderPath, fileName));
+    }
+
+    return fileNames;
+  }
+
+  /**
+   * Deletes a folder, including all of its contents.
+   * Behind the scenes is uses `fsx.removeSync()`
+   * @param folderPath - The absolute or relative path to the folder which should be deleted.
+   */
+  public static deleteFolder(folderPath: string): void {
+    fsx.removeSync(folderPath);
+  }
+
+  /**
+   * Deletes the content of a folder, but not the folder itself.
+   * Behind the scenes it uses `fsx.emptyDirSync()`
+   * @param folderPath - The absolute or relative path to the folder which should have its contents deleted.
+   */
+  public static emptyFolder(folderPath: string): void {
+    fsx.emptyDirSync(folderPath);
+  }
+
+  // ===============
+  // FILE OPERATIONS
+  // ===============
+
+  /**
+   * Writes a text string to a file on disk, overwriting the file if it already exists.
+   * Behind the scenes it uses `fs.writeFileSync()`
+   * @param filePath - The absolute or relative path of the file.
+   * @param contents - The text that should be written to the file.
+   * @param options - Optional settings that can change the behavior. Type: `IWriteFileOptions`
+   */
+  public static writeFile(filePath: string, contents: string, options?: IWriteFileOptions): void {
+    options = {
+      ensureFolder: false,
+      convertLineEndings: NewlineConversion.None,
+      encoding: 'utf8',
+      ...options
+    };
+
+    if (options.ensureFolder) {
+      const folderPath: string = pathUtilities.dirname(filePath);
+      FileSystem.createFolder(folderPath);
+    }
+
+    contents = FileSystem._convertLineEndings(contents, options.convertLineEndings);
+
+    fsx.writeFileSync(filePath, contents, { encoding: options.encoding });
+  }
+
+  /**
+   * Reads the contents of a file into a string.
+   * Behind the scenes it uses `fs.readFileSync()`.
+   * @param filePath - The path of the file to be read.
+   * @param options - Optional settings that can change the behavior. Type: `IReadFileOptions`
+   */
+  public static readFile(filePath: string, options?: IReadFileOptions): string {
+    options = {
+      encoding: 'utf8',
```

Comment:
>utf8 [](start = 17, length = 4)

The default encoding should be an exported cost.

Reply:
Huh? This is the default if you don't specify anything anyway.

---
In reply to: [198671550](https://github.com/Microsoft/web-build-tools/pull/727#discussion_r198671550) [](ancestors = 198671550)